src/main/scala/de/retest/guistatemachine/api/impl/serialization/GuiStateMachinGMLSerializer.scala

```diff
@@ -0,0 +1,68 @@
+package de.retest.guistatemachine.api.impl.serialization
+import java.io.{BufferedWriter, File, FileOutputStream, OutputStreamWriter}
+
+import com.github.systemdir.gml.YedGmlWriter
+import de.retest.guistatemachine.api.{GuiStateMachine, GuiStateMachineSerializer}
+import de.retest.recheck.ui.descriptors.SutState
+import org.jgrapht.graph.DirectedPseudograph
+
+case class GuiStateMachinGMLSerializer(guiStateMachine: GuiStateMachine) extends GuiStateMachineSerializer {
+
+  type GraphType = DirectedPseudograph[SutState, GraphActionEdge]
+
+  /**
+    * Converts the state machines into GML which can be read by editors like yED.
+    *
+    * @param filePath The file which the GML data is stored into.
+    * @throws RuntimeException If a vertex or edge cannot be added, this exception is thrown.
+    */
+  override def save(filePath: String): Unit = {
+    // get graph from user
+    val toDraw = createGraph()
+
+    // define the look and feel of the graph
+    val graphicsProvider = new GraphicsProvider
+
+    // get the gml writer
+    val writer =
+      new YedGmlWriter.Builder[SutState, GraphActionEdge, AnyRef](graphicsProvider, YedGmlWriter.PRINT_LABELS: _*)
+        .setEdgeLabelProvider(_.toString)
+        .setVertexLabelProvider(sutState => "%s - hash code: %d".format(sutState.toString, sutState.hashCode()))
+        .build
+
+    // write to file
+    val outputFile = new File(filePath)
+    val output = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(outputFile), "utf-8"))
+    try { writer.export(output, toDraw) } finally { if (output != null) { output.close() } }
+  }
+
+  override def load(filePath: String): Unit = throw new RuntimeException("Loading GML is not supported.")
```

Comment:
I think [`UnsupportedOperationException`](https://docs.oracle.com/javase/8/docs/api/?java/lang/UnsupportedOperationException.html) is better here.